var/spack/repos/builtin/packages/hpx/package.py

```diff
@@ -225,6 +225,11 @@ def instrumentation_args(self):
     def cmake_args(self):
         spec, args = self.spec, []
 
+        if "networking=mpi" in spec:
+            args.append(self.define("CMAKE_C_COMPILER", spec["mpi"].mpicc))
+            args.append(self.define("CMAKE_CXX_COMPILER", spec["mpi"].mpicxx))
+            args.append(self.define("CMAKE_Fortran_COMPILER", spec["mpi"].mpifc))
+
```

Comment:
Is this not going to conflict with the `CMAKE_CXX_COMPILER` set when using `+rocm`?

Most of the time, but not always, CMake not finding an external MPI can be fixed by changing the package configuration for that system. Is it possible that your external `cray-mpich` definition is missing some modules that are required to have the package work? If you provide some more details on the failure and how you've set things up we might be able to find an alternative solution.

Setting the `CMAKE_CXX_COMPILER` to `hipcc` with `+rocm` is also something that HPX could move away from, but that requires upstream changes and would only apply from the next version.